lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -285,10 +285,10 @@ def generate
                                 ColumnRename
                                 ColumnChange
                                   PrimaryKeyChange
-                                  IndexAdd
-                                    ForeignKeyAdd
-                                    ForeignKeyRemove
-                                  IndexRemove
+                                  ForeignKeyRemove
+                                    IndexRemove
+                                    IndexAdd
+                                  ForeignKeyAdd
```

Comment:
I realized that Remove-then-Add is necessary because indexes/constraints are named and SQL enforces that the names be unique. Also foreign keys depend on indexes so it makes the most sense to remove the foreign keys before removing indexes.